packages/dd-trace/src/appsec/reporter.js

```diff
@@ -82,12 +103,89 @@ function filterHeaders (headers, map) {
   return result
 }
 
-function formatHeaderName (name) {
-  return name
-    .trim()
-    .slice(0, 200)
-    .replace(/[^a-zA-Z0-9_\-:/]/g, '_')
-    .toLowerCase()
+function filterExtendedHeaders (headers, excludedHeaderNames, tagPrefix, limit = 0) {
+  const result = {}
+
+  if (!headers) return result
+
+  let counter = 0
+  for (const [headerName, headerValue] of Object.entries(headers)) {
+    if (counter >= limit) break
+    if (!excludedHeaderNames.includes(headerName)) {
+      result[getHeaderTag(tagPrefix, headerName)] = '' + headerValue
+      counter++
+    }
+  }
+
+  return result
+}
+
+function getCollectedHeaders (req, res, shouldCollectEventHeaders) {
+  // Mandatory
+  const mandatoryCollectedHeaders = filterHeaders(req.headers, REQUEST_HEADERS_MAP)
+
+  // Basic collection
+  if (!shouldCollectEventHeaders) return mandatoryCollectedHeaders
+
+  const requestEventCollectedHeaders = filterHeaders(req.headers, EVENT_HEADERS_MAP)
+  const responseEventCollectedHeaders = filterHeaders(res.getHeaders(), RESPONSE_HEADERS_MAP)
+
+  // Extended collection
+  if (!(extendedCollection?.enabled && !extendedCollection?.redaction)) {
+    return Object.assign(
+      mandatoryCollectedHeaders,
+      requestEventCollectedHeaders,
+      responseEventCollectedHeaders
+    )
+  }
+
+  const requestExtendedHeadersAvailableCount =
```

Comment:
The comment is making me confuse. The extended collection starts on line 142 and not here, the code into the `if` is the "standard" collection. 
```suggestion
  if (!(extendedCollection?.enabled && !extendedCollection?.redaction)) {
    // standard collection
    return Object.assign(
      mandatoryCollectedHeaders,
      requestEventCollectedHeaders,
      responseEventCollectedHeaders
    )
  }

  // Extended collection
  const requestExtendedHeadersAvailableCount =
```